File: chisel_py/src/chisel/checker/rule_metadata.py
PYTHON_BACKEND_SKILL = "building-python-backend"
SVELTEKIT_FRONTEND_SKILL = "building-sveltekit-frontend"
SVELTE_UI_SKILL = "designing-svelte-ui"
QA_SKILL = "qa"


def skill_name_for_category(category: str) -> str:
    if category == "test-structure":
        return QA_SKILL
    if category in {"component-enforcement", "colour", "responsiveness"}:
        return SVELTE_UI_SKILL
    if category in {"api", "api-endpoints"}:
        return SVELTEKIT_FRONTEND_SKILL
    return PYTHON_BACKEND_SKILL


def skill_name_for_rule(rule_id: str) -> str:
    category = rule_id.split(":", maxsplit=1)[0]
    return skill_name_for_category(category)

File: chisel_js/src/chisel/checker/rule_metadata.ts
export const SVELTEKIT_FRONTEND_SKILL = "building-sveltekit-frontend";
export const SVELTE_UI_SKILL = "designing-svelte-ui";
export const QA_SKILL = "qa";

export interface RuleInfo {
  readonly id: string;
  readonly category: string;
  readonly description: string;
  readonly fixGuidance: string;
  readonly skillName?: string;
}

export function skillNameForCategory(category: string): string {
  if (category === "test-structure") return QA_SKILL;
  if (["component-enforcement", "colour", "responsiveness"].includes(category)) {
    return SVELTE_UI_SKILL;
  }
  return SVELTEKIT_FRONTEND_SKILL;
}

export function skillNameForRule(ruleId: string): string {
  return skillNameForCategory(ruleId.split(":", 1)[0] ?? ruleId);
}

export function withSkillName(rule: RuleInfo): RuleInfo & { readonly skillName: string } {
  return {
    ...rule,
    skillName: rule.skillName ?? skillNameForCategory(rule.category),
  };
}
